factory/builder.py

```diff
@@ -225,9 +225,16 @@ def chain(self):
         return (self.stub,) + parent_chain
 
     def recurse(self, factory, declarations, force_sequence=None):
+        if not hasattr(factory, '_meta') or not hasattr(factory._meta, 'factory'):
+            raise errors.AssociatedClassError(
+                "%r: Attempting to recursing into a non-factory object %r"
+                % (self, factory))
         builder = self.builder.recurse(factory._meta, declarations)
         return builder.build(parent_step=self, force_sequence=force_sequence)
 
+    def __repr__(self):
+        return "<BuildStep for %r>" % (self.builder,)
```

Comment:
```suggestion
        return "<BuildStep for %r>" % self.builder
```

Reply:
I disagree: `self.builder` might be iterable, in which case this will break weirdly. I could use `.format()`, but I can't remember the marker for `repr` :smiling_imp:

Reply:
I fail to see when `self.builder` would not be an iterable? Its only caller is https://github.com/francoisfreitag/factory_boy/blob/59ee6584d8788af065a86aa168e8f42a4e9f3801/factory/builder.py#L263-L267, where builder is a `StepBuilder`.